shared/src/main/scala/bloop/io/Paths.scala

```diff
@@ -80,7 +82,24 @@ object Paths {
     out.toList
   }
 
-  case class AttributedPath(path: AbsolutePath, lastModifiedTime: FileTime, size: Long)
+  // sealed abstract is an abomination for scala 2.12 to get private which actually works
+  sealed abstract case class AttributedPath(
```

Comment:
Why change it to an abstract class?

Reply:
What the comment says really. To force all usage to go through the `of` constructor method. 

Older scala versions and private case class constructors is a surprisingly complex topic. For instance this article touches on it https://tuleism.github.io/blog/2020/scala-smart-constructors/ 

That said, now that the compiler showed me all usage i can change it back to something more normal if this is too weird. I just like having the guarantee

Reply:
Ach, sorry! I saw that previously when checking and then forgot :sweat: